src/main/scala/reactivemongo/json/collection/jsoncollection.scala

```diff
@@ -186,13 +164,7 @@ object JSONBatchCommands
   implicit object UpsertedReader extends pack.Reader[Upserted] {
     def reads(js: JsValue): JsResult[Upserted] = for {
       ix <- (js \ "index").validate[Int]
-      id <- (js \ "_id") match {
-        case _: JsUndefined =>
-          JsError(__ \ "_id", "error.objectId")
-
-        case js =>
-          BSONFormats.toBSON(js)
-      }
+      id <- (js \ "_id").validate[JsValue].flatMap(BSONFormats.toBSON)
```

Comment:
So my process for migrating quite a lot of this was to use the mainline reactivemongo repository as a guide - as it is compatible with play 2.4/2.5.
In this case I suspect in this case they assume the _id filed is always present so have removed the error case.